shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegateTest.m

```diff
@@ -22,51 +21,71 @@ - (void)testCreate {
 }
 
 - (void)testDidEnterBackground {
+  XCTNSNotificationExpectation* expectation = [[XCTNSNotificationExpectation alloc]
+      initWithName:UIApplicationDidEnterBackgroundNotification];
   FlutterPluginAppLifeCycleDelegate* delegate = [[FlutterPluginAppLifeCycleDelegate alloc] init];
   id plugin = OCMProtocolMock(@protocol(FlutterPlugin));
   [delegate addDelegate:plugin];
   [[NSNotificationCenter defaultCenter]
       postNotificationName:UIApplicationDidEnterBackgroundNotification
                     object:nil];
+
+  [self waitForExpectations:@[ expectation ] timeout:5.0];
   OCMVerify([plugin applicationDidEnterBackground:[UIApplication sharedApplication]]);
 }
 
 - (void)testWillEnterForeground {
+  XCTNSNotificationExpectation* expectation = [[XCTNSNotificationExpectation alloc]
+      initWithName:UIApplicationWillEnterForegroundNotification];
+
   FlutterPluginAppLifeCycleDelegate* delegate = [[FlutterPluginAppLifeCycleDelegate alloc] init];
   id plugin = OCMProtocolMock(@protocol(FlutterPlugin));
   [delegate addDelegate:plugin];
   [[NSNotificationCenter defaultCenter]
       postNotificationName:UIApplicationWillEnterForegroundNotification
                     object:nil];
+  [self waitForExpectations:@[ expectation ] timeout:5.0];
   OCMVerify([plugin applicationWillEnterForeground:[UIApplication sharedApplication]]);
 }
 
-- (void)skip_testWillResignActive {
+- (void)testWillResignActive {
+  XCTNSNotificationExpectation* expectation =
+      [[XCTNSNotificationExpectation alloc] initWithName:UIApplicationWillResignActiveNotification];
+
   FlutterPluginAppLifeCycleDelegate* delegate = [[FlutterPluginAppLifeCycleDelegate alloc] init];
   id plugin = OCMProtocolMock(@protocol(FlutterPlugin));
   [delegate addDelegate:plugin];
   [[NSNotificationCenter defaultCenter]
       postNotificationName:UIApplicationWillResignActiveNotification
                     object:nil];
+  [self waitForExpectations:@[ expectation ] timeout:5.0];
```

Comment:
It's failing on the `-[NSNotificationCenterpostNotificationName::]` line before it even gets here.
```
	0   CoreFoundation                      0x000000010e14de6e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x000000010d5b19b2 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010e14dcac +[NSException raise:format:] + 188
	3   libocmock_shared.dylib              0x0000000126ec2448 -[OCMockObject handleInvocation:] + 39
	4   libocmock_shared.dylib              0x0000000126ec2327 -[OCMockObject forwardInvocation:] + 30
	5   CoreFoundation                      0x000000010e152616 ___forwarding___ + 838
	6   CoreFoundation                      0x000000010e154c58 __forwarding_prep_1___ + 120
	7   libios_test_flutter.dylib           0x0000000126f4aa5d -[FlutterViewController surfaceUpdated:] + 187
	8   libios_test_flutter.dylib           0x0000000126f4b59d -[FlutterViewController applicationWillResignActive:] + 46
	9   CoreFoundation                      0x000000010e078d2c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	10  CoreFoundation                      0x000000010e0781a5 _CFXRegistrationPost1 + 421
	11  CoreFoundation                      0x000000010e077f11 ___CFXNotificationPost_block_invoke + 193
	12  CoreFoundation                      0x000000010e175473 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1795
	13  CoreFoundation                      0x000000010e077866 _CFXNotificationPost + 950
	14  Foundation                          0x000000010cff426b -[NSNotificationCenter postNotificationName:object:userInfo:] + 59
	15  libios_test_flutter.dylib           0x0000000126ef9932 -[FlutterPluginAppLifeCycleDelegateTest testWillResignActive] + 200
```
`Cannot use mock object OCClassMockObject(FlutterEngine)`
Where even is that `FlutterEngine` mock even coming from?  I see a `OCMClassMock([FlutterEngine class])` set up in `FlutterTextInputPluginTest` and `FlutterViewControllerTest` but not this test.  Maybe the view controllers in those tests are still alive after `-tearDown and are trying to notify their dead engines?  Is there a retain cycle somewhere?

Reply:
Yea, it's a leak from another test somewhere.  I'm guessing there is a stray mock engine that is getting talked to when the notification is sent (since the flutterengine also listens to those notifications).  I gotta add a `stopMocking` somewhere, or maybe see where it isn't removing itself as an observer when it should be.  The retain cycle might be created by not calling `stopMocking` and not necessarily a problem in the code.  I'll see what I can find.

Reply:
Found it, it was the partial mock in `testCallsNotifyLowMemory`.  I added a few extra stopMocking calls for FlutterEngines.

Reply:
I'm not sure that's the problem, the error is complaining that `stopMocking` had already been called on the engine, but a `FlutterViewController` is trying to tell it about `surfaceUpdated`.
```
Cannot use mock object OCClassMockObject(FlutterEngine) at 0x6000002144b0. This error usually occurs when a mock object is used after stopMocking has been called on it.
```
The only two `OCMClassMock([FlutterEngine class])` I see are in `FlutterTextInputPluginTest` and `FlutterViewControllerTest`.  Are the `FlutterViewController`s in `FlutterViewControllerTest` dealloced after `-tearDown`?  It doesn't look like it should be since the `self.mockEngine` is nil'd out in `-tearDown` which should break the retain cycle, but maybe I'm missing something.
```
  FlutterViewController* viewControllerB =
      [[FlutterViewController alloc] initWithEngine:self.mockEngine nibName:nil bundle:nil];
...
OCMStub([self.mockEngine viewController]).andReturn(viewControllerB);
```
Might be easiest to check if the memory graph contains any `FlutterViewController`s  before `FlutterPluginAppLifeCycleDelegateTest` starts.

Reply:
I've seen this problem because the mock object can cause retains on objects that aren't apparent, stopMocking flushes them out.  That was the thesis behind the change.  I saw those other tests that deleting the engine, that may be the problem.  Let me try that out.